### PR TITLE
MB- 15179-Reason-required-For-All- SIT

### PR DESCRIPTION
**Pull request by NamibiaTorres**
## Summary

The purpose of the ticket is to ensure that reason is required for all SIT service items (destination and origin sit). In the API we have made reason required in the origin site, but not in destination SIT. `Prime.yaml` has been updated to reflect this correction, and validation has been added in the `payload_to_model` for both origin and destination SIT. The `reason` field has also been added where it was previously missing in other files.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### How to test

1. Check that the prime.yaml, Payload_to_model, and model_to_payload have been updated for SIT service items
2. Run the test in tests:
```
go test ./pkg/handlers/primeapi/payloads --testify.m TestMTOServiceItemModel -v
```

```
go test ./pkg/handlers/primeapi--testify.m TestCreateMTOServiceItemHandler -v
```

```
go test ./pkg/handlers/primeapi--testify.m TestCreateMTOServiceItemDestSITHandler -v
```

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/about/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

### CODE REVIEW

File: pkg/handlers/primeapi/payloads/payload_to_model.go [hunk 1]
```diff
@@ -608,3 +631,23 @@ func validateDDFSIT(m primemessages.MTOServiceItemDestSIT) *validate.Errors {
...
+func validateReasonDestSIT(m primemessages.MTOServiceItemDestSIT) *validate.Errors {
+	verrs := validate.NewErrors()
+
+	if m.Reason == nil {
+		verrs.Add("reason", "reason is required in body.")
+	}
+	return verrs
+}
+
+// validateReasonOriginSIT validates that Destination SIT service items have required Reason field
```
Comment (kctruss) [inline pkg/handlers/primeapi/payloads/payload_to_model.go:645]: **nitpick:** should be "validates that Origin SIT . . ." ‏

File: pkg/handlers/primeapi/payloads/payload_to_model.go [hunk 2]
```diff
@@ -608,3 +631,23 @@ func validateDDFSIT(m primemessages.MTOServiceItemDestSIT) *validate.Errors {
...
+func validateReasonOriginSIT(m primemessages.MTOServiceItemOriginSIT) *validate.Errors {
+	verrs := validate.NewErrors()
+
+	if m.Reason == nil {
```
Comment (kctruss) [inline pkg/handlers/primeapi/payloads/payload_to_model.go:649]: **suggestion:** Same as above: `m.Reason == nil OR m.Reason == ""`‏

**Review by kctruss — COMMENTED** [review]
Great work on this @NamibiaTorres ! Just a few comments with suggestions for testing and validation.

I tested using Postman and hitting the `createMTOServiceItem` Prime API endpoint directly.
When I sent a request with no `reason` value, I received the expected error. Updating my payload with a `reason` value works as expected.

I do want to call out that when I sent a payload that included the value `"reason": ""` (empty string), instead of the validation error I received a 500 error. This is likely related to how errors are returned from the data layer, but could be resolved by adding an empty string check earlier alongside the nil check.

File: pkg/handlers/primeapi/payloads/payload_to_model.go (hunk 1 above)
Comment (NamibiaTorres) [reply pkg/handlers/primeapi/payloads/payload_to_model.go:645]: Yes, you're right! I'll make the correction.

File: pkg/handlers/primeapi/payloads/payload_to_model.go [hunk 3]
```diff
@@ -608,3 +631,23 @@ func validateDDFSIT(m primemessages.MTOServiceItemDestSIT) *validate.Errors {
 	}
 	return verrs
 }
+
+// validateReasonDestSIT validates that Destination SIT service items have required Reason field
+func validateReasonDestSIT(m primemessages.MTOServiceItemDestSIT) *validate.Errors {
+	verrs := validate.NewErrors()
+
+	if m.Reason == nil {
```
[the remark this answers is not in the document]
Comment (NamibiaTorres) [reply pkg/handlers/primeapi/payloads/payload_to_model.go:639]: That's a great suggestion! Thank you!

**Review by kctruss — APPROVED** [review]
highfive

**Pull request merged**